scripts/gulp/plugins/util/docgen.ts

```diff
@@ -318,6 +353,144 @@ export class Parser {
     return null;
   }
 
+  private static isLiteral(type: ResolvedType): boolean {
+    return type.type === 'StringLiteral' || type.type === 'NumberLiteral' || type.type === 'BooleanLiteral';
+  }
+
+  private customResolveTypeOverride(typeNode, name): ResolvedType | undefined {
+    if (typeNode?.symbol?.parent?.getEscapedName?.() === 'React') {
+      const symbolName = typeNode.symbol.getEscapedName?.();
+      if (symbolName === 'ReactElement') {
+        return {
+          type: 'ReactElement',
+          name,
+        };
+      } else if (symbolName === 'ReactPortal') {
+        return {
+          type: 'ReactPortal',
+          name,
+        };
+      }
+    }
+
+    if (
+      name === 'ICSSInJSStyle' || // expanding this leads to a single component.info being MBs big
+      name === 'Document' || // OOM :-/
+      name === 'HTMLElement'
+    ) {
+      return {
+        type: 'object',
+        props: {},
+        name,
+      } as RTObject;
+    }
+
+    if (name === 'Document') {
+      return {
+        type: 'object',
+        props: {},
+        name,
+      } as RTObject;
+    }
+
+    return undefined;
+  }
+
+  private resolveType(typeNode, depth = 0): ResolvedType {
+    const name = this.checker.typeToString(typeNode);
+
+    const customType = this.customResolveTypeOverride(typeNode, name);
+    if (customType) {
+      return customType;
+    }
+
+    if (depth > 5) {
```

Comment:
Avoid huge info files, we don't need deep for unions
```suggestion
    if (depth > 1) {
```